Azure/sonic-buildimage
# MCLAG enhacements ICCPd initial code commit

- Why I did it
This PR contains the ICCPd changes to support MCLAG enhancements feature. 
- How I did it
Implemented as per the MCLAG enhancements HLD mentioned at :  https://github.com/Azure/SONiC/pull/596
- How to verify it
Compile and verify MCLAG behavior. 
- Description for the changelog
The PR includes the ICCPD code changes for the MCLAG enhancements support.
This PR must work with submitted PR's in other sonic repositories which are listed below. 
    > Swss: https://github.com/Azure/sonic-swss/pull/1331                                                                    
    > swss-common: https://github.com/Azure/sonic-swss-common/pull/405 
    > Mclagsyncd: https://github.com/Azure/sonic-swss/pull/1349                                           
    > Utilities: https://github.com/Azure/sonic-utilities/pull/1138 
    > Mgmt-FW: https://github.com/Azure/sonic-mgmt-framework/pull/59 
    > Mgmt-Common: https://github.com/Azure/sonic-mgmt-common/pull/25
